packages/shared-data/types/provider-config.ts

```diff
@@ -0,0 +1,56 @@
+import { Type, type Static } from "@sinclair/typebox";
+
+import supportedModels from "../json/supported-models.json";
+
+export const ProviderNameEnum = Type.Enum(
+  Object.fromEntries(
+    [
+      ...new Set(
+        supportedModels.flatMap((model) =>
+          (model.providers ?? []).flatMap((providerObj) =>
+            Object.keys(providerObj),
+          ),
+        ),
+      ),
+    ].map((providerName) => [providerName, providerName]),
+  ),
+  { error: "Invalid provider name" },
+);
+
+const AwsProviderConfigSchema = Type.Object({
+  bedrockRoleArn: Type.String(),
+  region: Type.String(),
+  baseURL: Type.Optional(Type.String()),
+});
+
+const GoogleProviderConfigSchema = Type.Object({
+  serviceAccountEmail: Type.String(),
+  audience: Type.String(),
+  location: Type.String(),
+  project: Type.String(),
+  baseURL: Type.Optional(Type.String()),
+});
+
+const ApiKeyProviderConfigSchema = Type.Object({
+  apiKey: Type.String({ "x-redact": true }),
+  baseURL: Type.Optional(Type.String()),
+});
+
+export const ProviderConfigConfig = Type.Union([
+  AwsProviderConfigSchema,
+  GoogleProviderConfigSchema,
+  ApiKeyProviderConfigSchema,
```

Comment:
For what is this?

Reply:
Does it work for Groq in this PR? Or is there additional work needed to support Groq?